qemu/tests/chardev_tls_encryption.py

```diff
@@ -0,0 +1,94 @@
+import aexpect
+
+from avocado.utils import process
+
+from virttest import utils_misc
+from virttest import env_process
+from virttest import error_context
+
+from provider.chardev_utils import setup_certs
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test native TLS encryption on chardev TCP transports
+    Scenario 1:
+        a. Run gnutls server
+        b. Launch QEMU with a serial port as TLS client
+        c. Check the server endpoint output
+    Scenario 2:
+        a. Launch QEMU with a serial port as TLS server
+        b. Run gnutls client to connect TLS server
+        c. Check the client endpoint output
+    Scenario 3:
+        a. Launch QEMU with a serial port as TLS server
+        b. Execute 'cat /dev/ttyS0' in guest which boot from step 1
+        c. Launch QEMU with a serial port as TLS client
+        d. Check the output of step b
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    clean_cmd = params["clean_cmd"]
+    try:
+        setup_certs(params)
+        hostname = process.run('hostname', ignore_status=False, shell=True,
+                               verbose=True).stdout_text.strip()
+        port = str(utils_misc.find_free_ports(5000, 9999, 1, hostname)[0])
+
+        # Scenario 1
+        gnutls_cmd_server = params.get("gnutls_cmd_server")
+        if gnutls_cmd_server:
+            gnutls_cmd_server = gnutls_cmd_server % port
+            params["extra_params"] = params["extra_params"] % (hostname, port)
+            tls_server = aexpect.run_bg(gnutls_cmd_server)
```

Comment:
From debug.log, can not get the full steps clearly, so suggest add some context when start gnutls server and client~